### PR TITLE
Fix Search Panel

### PR DESCRIPTION
**Pull request by jeffreye**
* Fix no match popup
* Fix Bindings
* Fix text input

### CODE REVIEW

File: src/AvaloniaEdit/Search/SearchPanel.xaml [hunk 1]
```diff
@@ -37,8 +37,17 @@
...
+                               StaysOpen="True"
+                               Focusable="False"
+                               IsOpen="False"
+                               ObeyScreenEdges="True">
+                            <ContentControl>Message Here</ContentControl>
+                            </Popup>
                         <ToggleButton Classes="ExpanderToggle" Name="Expander" VerticalAlignment="Top"
-                                      IsChecked="{TemplateBinding IsReplaceMode, Mode=TwoWay}"
+                                      IsChecked="{Binding IsReplaceMode, Mode=TwoWay, RelativeSource={RelativeSource TemplatedParent}}"
                                       Grid.Column="0"
```
Comment (aelij) [inline src/AvaloniaEdit/Search/SearchPanel.xaml]: If you just needed a two way binding, Avalonia has Mode in TemplateBinding

File: src/AvaloniaEdit/Search/SearchPanel.xaml [hunk 2]
```diff
@@ -37,8 +37,17 @@
...
                     <Grid ColumnDefinitions="Auto,Auto,Auto"
                 RowDefinitions="Auto,Auto">
+                        <Popup Name="PART_MessageView"
+                               Classes="ToolTip"
+                               PlacementMode="Bottom"
+                               StaysOpen="True"
+                               Focusable="False"
+                               IsOpen="False"
+                               ObeyScreenEdges="True">
+                            <ContentControl>Message Here</ContentControl>
```
Comment (aelij) [inline src/AvaloniaEdit/Search/SearchPanel.xaml]: I think a `ContentPresenter` is good enough here. Also, add a named PART for the presenter, right now the code assumes a specific visual tree.

File: src/AvaloniaEdit/Search/SearchPanel.cs [hunk 3]
```diff
@@ -456,8 +451,6 @@ private void SearchLayerKeyDown(object sender, KeyEventArgs e)
                     Close();
                     break;
             }
-
-            e.Handled = true;
```
Comment (aelij) [inline src/AvaloniaEdit/Search/SearchPanel.cs]: Why this change?

Comment (jeffreye) [reply src/AvaloniaEdit/Search/SearchPanel.cs]: In previous version, Search panel doesn't not take any input from keyboard

File: src/AvaloniaEdit/Search/SearchPanel.xaml (hunk 1 above)
Comment (jeffreye) [reply src/AvaloniaEdit/Search/SearchPanel.xaml]: It was template binding previously, but it doesn't work.

File: src/AvaloniaEdit/Search/SearchPanel.xaml (hunk 2 above)
Comment (jeffreye) [reply src/AvaloniaEdit/Search/SearchPanel.xaml]: Do your you mean use `ContentPresenter` instead of `ContentControl`?

Comment (aelij) [reply src/AvaloniaEdit/Search/SearchPanel.xaml]: Yes

**Comment by aelij** [comment]
Thanks for the fix @jeffreye
